tgui/packages/tgui/interfaces/BotAnnouncement.tsx

```diff
@@ -0,0 +1,270 @@
+  string: string;
+  index: number;
+};
+
+type BotAnnouncementData = {
+  channels: string[];
+  lines: string[];
+  button_data: ButtonData[];
+  cooldown_left: number;
+};
+
+enum TAB {
+  Announcements,
+  Shortcuts,
+}
+
+export const BotAnnouncement = (props) => {
+  const { act, data } = useBackend<BotAnnouncementData>();
+  const { channels, lines, button_data, cooldown_left } = data;
+
+  const [tab, setTab] = useState(TAB.Announcements);
+  const [selectedChannel, setSelectedChannel] = useState<null | string>(null);
+  const [selectedLine, setSelectedLine] = useState<null | number>(null);
+  const [selectedButton, setSelectedButton] = useState<null | number>(null);
+  const [search, setSearch] = useState('');
+
+  let filteredLines: StringWithId[] = lines.map((val, index) => ({
+    string: val,
+    index,
+  }));
+  let filteredShortcuts: ButtonDataWithId[] = button_data.map((val, index) => ({
+    button: val,
+    index,
+  }));
+
+  if (search !== '') {
+    if (tab === TAB.Announcements) {
+      const lineSearch = createSearch(
+        search,
+        (item: StringWithId) => item.string,
+      );
+      filteredLines = filteredLines.filter(lineSearch);
+    } else {
+      const buttonSearch = createSearch(
+        search,
+        (item: ButtonDataWithId) => item.button.name,
+      );
+      filteredShortcuts = filteredShortcuts.filter(buttonSearch);
+    }
+  }
+
+  return (
+    <Window width={650} height={500}>
+      <Window.Content>
+        <Section fitted m={0}>
+          <Tabs>
+            <Tabs.Tab
+              selected={tab === TAB.Announcements}
+              onClick={() => {
+                setSearch('');
+                setTab(TAB.Announcements);
+              }}
+            >
+              Announcements
+            </Tabs.Tab>
+            <Tabs.Tab
+              selected={tab === TAB.Shortcuts}
+              onClick={() => {
+                setSearch('');
+                setTab(TAB.Shortcuts);
+              }}
+            >
+              Shortcuts
+            </Tabs.Tab>
+          </Tabs>
+        </Section>
+        <Section m={0} scrollable fill maxHeight="350px">
+          {tab === TAB.Announcements ? (
+            <Stack vertical zebra>
+              {filteredLines.map((val) => (
+                <Stack.Item key={val.string}>
+                  <Button
+                    py={1}
+                    color={selectedLine === val.index ? 'green' : 'transparent'}
+                    fluid
+                    onClick={() => setSelectedLine(val.index)}
+                    minHeight="32px"
+                  >
+                    <Stack>
+                      <Stack.Item>
+                        <Icon name="volume-high" />
+                      </Stack.Item>
+                      <Stack.Item>{val.string}</Stack.Item>
+                    </Stack>
+                  </Button>
+                </Stack.Item>
+              ))}
+            </Stack>
+          ) : null}
```

Comment:
my last comment might have been misleading, sorry. there's no need use a ternary if null is the second case. use a `tab === TAB.Announcements && <Component/>` for this and the next occurrence.
ternaries only if you have a `someBool ? <ComponentA/> : <ComponentB/>` scenario.